File: app/src/main/java/hu/andras/daggersample/ui/MainActivity.java
package hu.andras.daggersample.ui;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;

import androidx.appcompat.app.AppCompatActivity;

import hu.andras.daggersample.R;
import hu.andras.daggersample.ui.feature1.list.Feature1ListActivity;
import hu.andras.daggersample.ui.feature2.list.Feature2ListActivity;
import hu.andras.daggersample.ui.feature3.list.Feature3ListActivity;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
    }

    public void startComponentDependency(View view) {
        startActivity(new Intent(this, Feature1ListActivity.class));
    }

    public void startSubcomponent(View view) {
        startActivity(new Intent(this, Feature2ListActivity.class));
    }

    public void startAndroidInjection(View view) {
        startActivity(new Intent(this, Feature3ListActivity.class));
    }
}

File: app/src/main/java/hu/andras/daggersample/ui/MainActivity.kt
package hu.andras.daggersample.ui

import android.content.Intent
import android.os.Bundle
import android.view.View
import androidx.appcompat.app.AppCompatActivity
import hu.andras.daggersample.R
import hu.andras.daggersample.ui.feature1.list.Feature1ListActivity
import hu.andras.daggersample.ui.feature2.list.Feature2ListActivity

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
    }

    fun startComponentDependency(view: View?) {
        startActivity(Intent(this, Feature1ListActivity::class.java))
    }

    fun startSubcomponent(view: View?) {
        startActivity(Intent(this, Feature2ListActivity::class.java))
    }
}
